b1ddi/provider.go

```diff
@@ -147,3 +147,32 @@ func dataSourceSchemaFromResource(resource func() *schema.Resource) *schema.Reso
 		Schema: resultSchema,
 	}
 }
+
+// dataSourceSchemaSubnetFromResource -- generates schema for results field in the subnet and next_available_subnet data source
+// This function gets the original resource, schema
+// - updates the parent and address field schema to remove 'ExactlyOneOf'
+// - injects the ID field in it.
+func dataSourceSchemaSubnetFromResource(resource func() *schema.Resource) *schema.Resource {
+	// Get the resource schema
+	resultSchema := dataSourceSchemaFromResource(resource).Schema
+	resultSchema["parent"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "The resource identifier.",
+	}
+	resultSchema["address"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "The address of the subnet in the form “a.b.c.d/n” where the “/n” may be omitted. In this case, the CIDR value must be defined in the _cidr_ field. When reading, the _address_ field is always in the form “a.b.c.d”.",
+	}
+
+	// Inject id field into the resource schema
+	resultSchema["id"] = &schema.Schema{
```

Comment:
Isn't it a no-op considering what's done on line 141?

Reply:
Yes, it is. Will fix it